Grafana 2.0.0 -> 2.1.0 panic during "DB Migration"

Grafana 2.0.0 -> 2.1.0 upgrade attempt results in panic when attempting "DB Migration" with PG backend for dashboards & session state.

```
# ./grafana
2015/06/26 11:12:14 [I] Starting Grafana
2015/06/26 11:12:14 [I] Version: master, Commit: NA, Build date: 1969-12-31 16:00:00 -0800 PST
2015/06/26 11:12:14 [I] Configuration Info
Config files:
  [0]: /opt/go/src/github.com/grafana/grafana/conf/defaults.ini
Paths:
  home: /opt/go/src/github.com/grafana/grafana
  data: /opt/go/src/github.com/grafana/grafana
  logs: /opt/go/src/github.com/grafana/grafana

2015/06/26 11:12:14 [I] Database: postgres, ConnectionString: user=grafana password=pass123 host=dbserver1 port=5432 dbname=grafana_dashboards sslmode=disable
2015/06/26 11:12:14 [I] Migrator: Starting DB migration
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x7ea4ee]

goroutine 1 [running]:
gopkg.in/ini%2ev1.(*Section).GetKey(0x0, 0xbf07d0, 0x7, 0x0, 0x0, 0x0)
    /opt/twitch/go/src/gopkg.in/ini.v1/ini.go:484 +0x4e
gopkg.in/ini%2ev1.(*Section).Key(0x0, 0xbf07d0, 0x7, 0x0)
    /opt/twitch/go/src/gopkg.in/ini.v1/ini.go:502 +0x3c
github.com/grafana/grafana/pkg/services/search.Init()
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/services/search/handlers.go:18 +0xad
main.main()
    /opt/twitch/go/src/github.com/grafana/grafana/main.go:56 +0xdb

goroutine 5 [select]:
github.com/grafana/grafana/pkg/log.(*Logger).StartLogger(0xc20802d740)
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:223 +0x369
created by github.com/grafana/grafana/pkg/log.newLogger
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:153 +0x13a

goroutine 6 [syscall]:
os/signal.loop()
    /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
    /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [chan receive]:
main.func·001()
    /opt/twitch/go/src/github.com/grafana/grafana/main.go:47 +0xed
created by main.main
    /opt/twitch/go/src/github.com/grafana/grafana/main.go:49 +0xc5

goroutine 8 [select]:
github.com/grafana/grafana/pkg/log.(*Logger).StartLogger(0xc20812ce00)
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:223 +0x369
created by github.com/grafana/grafana/pkg/log.newLogger
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:153 +0x13a

goroutine 9 [select]:
github.com/grafana/grafana/pkg/log.(*Logger).StartLogger(0xc20809a080)
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:223 +0x369
created by github.com/grafana/grafana/pkg/log.newLogger
    /opt/twitch/go/src/github.com/grafana/grafana/pkg/log/log.go:153 +0x13a

goroutine 17 [chan receive]:
database/sql.(*DB).connectionOpener(0xc2080fe1e0)
    /usr/local/go/src/database/sql/sql.go:589 +0x4c
created by database/sql.Open
    /usr/local/go/src/database/sql/sql.go:452 +0x31c

goroutine 33 [syscall, locked to thread]:
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
```

--- a/go.mod
+++ b/go.mod
@@ -53,7 +53,7 @@ require (
 	github.com/grafana/cuetsy v0.0.0-20211119211437-8c25464cc9bf
 	github.com/grafana/grafana-aws-sdk v0.10.1
 	github.com/grafana/grafana-azure-sdk-go v1.1.0
-	github.com/grafana/grafana-plugin-sdk-go v0.129.0
+	github.com/grafana/grafana-plugin-sdk-go v0.131.0
 	github.com/grafana/loki v1.6.2-0.20211015002020-7832783b1caa
 	github.com/grpc-ecosystem/go-grpc-middleware v1.3.0
 	github.com/hashicorp/go-hclog v0.16.1
@@ -272,7 +272,7 @@ require (
 	github.com/elazarl/goproxy v0.0.0-20220115173737-adb46da277ac // indirect
 	github.com/envoyproxy/go-control-plane v0.10.1 // indirect
 	github.com/envoyproxy/protoc-gen-validate v0.6.2 // indirect
-	github.com/getkin/kin-openapi v0.91.0 // indirect
+	github.com/getkin/kin-openapi v0.94.0 // indirect
 	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
 	github.com/golang-jwt/jwt v3.2.1+incompatible // indirect
 	github.com/imdario/mergo v0.3.12 // indirect

--- a/pkg/api/dashboard.go
+++ b/pkg/api/dashboard.go
@@ -453,9 +453,10 @@ func (hs *HTTPServer) GetHomeDashboard(c *models.ReqContext) response.Response {
 	dash.Meta.IsHome = true
 	dash.Meta.CanEdit = c.SignedInUser.HasRole(models.ROLE_EDITOR)
 	dash.Meta.FolderTitle = "General"
+	dash.Dashboard = simplejson.New()
 
 	jsonParser := json.NewDecoder(file)
-	if err := jsonParser.Decode(&dash.Dashboard); err != nil {
+	if err := jsonParser.Decode(dash.Dashboard); err != nil {
 		return response.Error(500, "Failed to load home dashboard", err)
 	}
 

--- a/pkg/api/dashboard_test.go
+++ b/pkg/api/dashboard_test.go
@@ -1095,8 +1095,8 @@ func restoreDashboardVersionScenario(t *testing.T, desc string, url string, rout
 }
 
 func (sc *scenarioContext) ToJSON() *simplejson.Json {
-	var result *simplejson.Json
-	err := json.NewDecoder(sc.resp.Body).Decode(&result)
+	result := simplejson.New()
+	err := json.NewDecoder(sc.resp.Body).Decode(result)
 	require.NoError(sc.t, err)
 	return result
 }

--- a/pkg/infra/usagestats/service/api_test.go
+++ b/pkg/infra/usagestats/service/api_test.go
@@ -73,11 +73,11 @@ func getUsageStats(t *testing.T, server *web.Mux) (*models.SystemStats, *httptes
 	recorder := httptest.NewRecorder()
 	server.ServeHTTP(recorder, req)
 
-	var usageStats *models.SystemStats
+	var usageStats models.SystemStats
 	if recorder.Code == http.StatusOK {
 		require.NoError(t, json.NewDecoder(recorder.Body).Decode(&usageStats))
 	}
-	return usageStats, recorder
+	return &usageStats, recorder
 }
 
 func setupTestServer(t *testing.T, user *models.SignedInUser, service *UsageStats) *web.Mux {

--- a/pkg/plugins/manager/signature/manifest.go
+++ b/pkg/plugins/manager/signature/manifest.go
@@ -82,7 +82,7 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 	}
 
 	// Convert to a well typed object
-	manifest := &pluginManifest{}
+	var manifest pluginManifest
 	err := json.Unmarshal(block.Plaintext, &manifest)
 	if err != nil {
 		return nil, errutil.Wrap("Error parsing manifest JSON", err)
@@ -99,7 +99,7 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 		return nil, errutil.Wrap("failed to check signature", err)
 	}
 
-	return manifest, nil
+	return &manifest, nil
 }
 
 func Calculate(mlog log.Logger, plugin *plugins.Plugin) (plugins.Signature, error) {

--- a/pkg/services/comments/commentmodel/models.go
+++ b/pkg/services/comments/commentmodel/models.go
@@ -56,9 +56,9 @@ func (s Settings) Value() (driver.Value, error) {
 func (s *Settings) Scan(value interface{}) error {
 	switch v := value.(type) {
 	case string:
-		return json.Unmarshal([]byte(v), &s)
+		return json.Unmarshal([]byte(v), s)
 	case []uint8:
-		return json.Unmarshal(v, &s)
+		return json.Unmarshal(v, s)
 	default:
 		return fmt.Errorf("type assertion on scan failed: got %T", value)
 	}

--- a/pkg/services/serviceaccounts/api/api.go
+++ b/pkg/services/serviceaccounts/api/api.go
@@ -179,7 +179,7 @@ func (api *ServiceAccountsAPI) updateServiceAccount(c *models.ReqContext) respon
 		return response.Error(http.StatusBadRequest, "Service Account ID is invalid", err)
 	}
 
-	cmd := &serviceaccounts.UpdateServiceAccountForm{}
+	var cmd serviceaccounts.UpdateServiceAccountForm
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "Bad request data", err)
 	}
@@ -188,7 +188,7 @@ func (api *ServiceAccountsAPI) updateServiceAccount(c *models.ReqContext) respon
 		return response.Error(http.StatusBadRequest, "Invalid role specified", nil)
 	}
 
-	resp, err := api.store.UpdateServiceAccount(c.Req.Context(), c.OrgId, scopeID, cmd)
+	resp, err := api.store.UpdateServiceAccount(c.Req.Context(), c.OrgId, scopeID, &cmd)
 	if err != nil {
 		switch {
 		case errors.Is(err, serviceaccounts.ErrServiceAccountNotFound):

--- a/pkg/web/webtest/webtest_test.go
+++ b/pkg/web/webtest/webtest_test.go
@@ -99,7 +99,7 @@ func TestServerClient(t *testing.T) {
 		require.NoError(t, err)
 		require.NoError(t, resp.Body.Close())
 
-		var user *models.SignedInUser
+		var user models.SignedInUser
 		err = json.Unmarshal(bytes, &user)
 		require.NoError(t, err)
 		require.NotNil(t, user)
@@ -121,7 +121,7 @@ func TestServerClient(t *testing.T) {
 		require.NoError(t, err)
 		require.NoError(t, resp.Body.Close())
 
-		var user *models.SignedInUser
+		var user models.SignedInUser
 		err = json.Unmarshal(bytes, &user)
 		require.NoError(t, err)
 		require.NotNil(t, user)

